From 81839e57d5f3d86d6663f58d5d6b7b0605a6986e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Klitzing?= <aklitzing@gmail.com>
Date: Tue, 30 Jul 2019 12:49:14 +0200
Subject: [PATCH] Add Q_CORE_EXPORT to lcEventDispatcher*

Undefined symbols for architecture arm64:
  "lcEventDispatcher()", referenced from:
      _qt_main_wrapper in libqios.a(qioseventdispatcher.o)
      +[QIOSApplicationStateTracker applicationDidFinishLaunching:] in libqios.a(qioseventdispatcher.o)
      user_main_trampoline() in libqios.a(qioseventdispatcher.o)
      +[QIOSApplicationStateTracker applicationWillTerminate] in libqios.a(qioseventdispatcher.o)
      QIOSJumpingEventDispatcher::interruptEventLoopExec() in libqios.a(qioseventdispatcher.o)
ld: symbol(s) not found for architecture arm64

Change-Id: I43677cf3bce0588753731d81d533f85b0ea1e223
Fixes: QTBUG-75457
---
 src/corelib/kernel/qeventdispatcher_cf.mm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git x/qtbase/src/corelib/kernel/qeventdispatcher_cf.mm y/qtbase/src/corelib/kernel/qeventdispatcher_cf.mm
index b7b379e2c1..ac711d74ee 100644
--- x/qtbase/src/corelib/kernel/qeventdispatcher_cf.mm
+++ y/qtbase/src/corelib/kernel/qeventdispatcher_cf.mm
@@ -148,8 +148,8 @@ static CFStringRef runLoopMode(NSDictionary *dictionary)
 
 QT_BEGIN_NAMESPACE
 
-Q_LOGGING_CATEGORY(lcEventDispatcher, "qt.eventdispatcher");
-Q_LOGGING_CATEGORY(lcEventDispatcherTimers, "qt.eventdispatcher.timers");
+Q_CORE_EXPORT Q_LOGGING_CATEGORY(lcEventDispatcher, "qt.eventdispatcher");
+Q_CORE_EXPORT Q_LOGGING_CATEGORY(lcEventDispatcherTimers, "qt.eventdispatcher.timers");
 
 class RunLoopDebugger : public QObject
 {
-- 
2.22.0

